core-bundle/src/Image/ImageHelper.php

```diff
@@ -0,0 +1,147 @@
+<?php
+
+declare(strict_types=1);
+
+/*
+ * This file is part of Contao.
+ *
+ * (c) Leo Feyer
+ *
+ * @license LGPL-3.0-or-later
+ */
+
+namespace Contao\CoreBundle\Image;
+
+use Contao\CoreBundle\Asset\ContaoContext;
+use Contao\CoreBundle\Framework\ContaoFramework;
+use Contao\FilesModel;
+use Contao\Frontend;
+use Contao\Image\PictureConfiguration;
+use Contao\Validator;
+use Symfony\Component\HttpFoundation\RequestStack;
+use Webmozart\PathUtil\Path;
+
+class ImageHelper
+{
+    public const PICTURE_IMAGE = 'img';
+    public const PICTURE_SOURCES = 'sources';
+    public const PICTURE_ALT = 'alt';
+
+    /**
+     * @var PictureFactoryInterface
+     */
+    private $pictureFactory;
+
+    /**
+     * @var string
+     */
+    private $rootDir;
+
+    /**
+     * @var string
+     */
+    private $staticUrl;
+
+    /**
+     * @var string
+     */
+    private $locale;
+
+    /**
+     * @var ContaoFramework
+     */
+    private $framework;
+
+    public function __construct(PictureFactoryInterface $pictureFactory, string $rootDir, ContaoContext $fileContext, RequestStack $requestStack, string $defaultLocale, ContaoFramework $framework)
+    {
+        $this->pictureFactory = $pictureFactory;
+        $this->rootDir = $rootDir;
+        $this->staticUrl = $fileContext->getStaticUrl();
+        $this->framework = $framework;
+
+        $request = $requestStack->getCurrentRequest();
+        $this->locale = null !== $request ? $request->getLocale() : $defaultLocale;
```

Comment:
Doesn’t this make the service stateful? What if the locale changes between the construction of the service and a call to `createPicture()`? Wouldn’t it be better to pass the locale directly to `createPicture`?

Reply:
You're right I guess. We need a way to pass the locale anyway (e.g. when running on CLI). Maybe we should still fallback to the request locale / default locale so that one does not need to provide it in most cases? Wdyt?

Reply:
You can just get the actual request from the request stack in the method you use the locale (just move it away from the constructor).

Reply:
Just saw, that `addImageToTemplate()` refers to `$objPage->language` / `$objPage->rootFallbackLanguage`. To recreate this behavior we should imo use the request to check for a page and follow the same logic if `createPicture()` wasn't called with an explicit locale.